### PR TITLE
Invalidate cached pages from PagingStore by fileId

### PR DESCRIPTION
**Pull request by huanghua78**
### What changes are proposed in this pull request?

Add and implement functions to invalidate all cached pages by fileId.

### Why are the changes needed?

When we know that some file is to be removed, or renamed, or overwritten, we need to invalidate the cached pages in PagingStore.
Please note, at this moment, it is possible that the target file is already removed from UFS. So, getStatus() of this file will fail with exception. But we still need to invalidate all the cached pages.

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java [hunk 1]
```diff
@@ -302,6 +302,14 @@ default List<PageId> getCachedPageIdsByFileId(String fileId, long fileLength) {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Invalidates all pages by its file id.
+   * @param fileId the file id of the target file
+   */
+  default void invalidateAllPagesByFileId(String fileId) {
+    // Do nothing;
+  }
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:311]: 1. Rename to `deleteFile` to be consistent with existing `deletePage` method, as `invalidation` is not something `CacheManager` understands. Also there is an existing `invalidate` method, which can be confused with this method.
2. Change to required method. 

```suggestion
  /**
   * Deletes all pages of the given file.
   * @param fileId the file id of the target file
   */
  void deleteFile(String fileId);
```

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultPageMetaStore.java [hunk 2]
```diff
@@ -227,6 +227,13 @@ PageInfo evictInternal(CacheEvictor evictor) {
     return victimInfo;
   }
 
+  @Override
+  @GuardedBy("getLock()")
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultPageMetaStore.java:231]:
```suggestion
  @GuardedBy("getLock().readLock()")
```

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 3]
```diff
@@ -801,6 +802,15 @@ public List<PageId> getCachedPageIdsByFileId(String fileId, long fileLength) {
     return pageIds;
   }
 
+  @Override
+  public void invalidateAllPagesByFileId(String fileId) {
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:806]: possible duplicate with the existing `invalidate` method.

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/PageMetaStore.java [hunk 4]
```diff
@@ -135,6 +137,15 @@ static PageMetaStore create(CacheManagerOptions options) throws IOException {
...
   void reset();
 
+  /**
+   * Gets all pages by the specified File ID.
+   * @param fileId the target file id
+   * @return a set of PageInfo's of this file, otherwise an empty set if not found
+   */
+  default Set<PageInfo> getAllPagesByFileId(String fileId) {
+    return Collections.emptySet();
+  }
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/PageMetaStore.java:147]: I think we can make this a required method.

```suggestion
  Set<PageInfo> getAllPagesByFileId(String fileId);
```

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java (hunk 1 above)
Comment (huanghua78) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:311]: OK.

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultPageMetaStore.java (hunk 2 above)
Comment (huanghua78) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultPageMetaStore.java:231]: OK

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java (hunk 3 above)
Comment (huanghua78) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:806]: Renamed to deleteFile() as suggested.

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/PageMetaStore.java (hunk 4 above)
Comment (huanghua78) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/PageMetaStore.java:147]: OK

**Comment by huanghua78** [comment]
@dbw9580 please review again.

**Review by dbw9580 — CHANGES_REQUESTED** [review]
LGTM except a minor issue

File: dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java [hunk 5]
```diff
@@ -354,25 +354,8 @@ public boolean invalidateCachedFile(FileInfo fileInfo) {
   }
 
   private boolean invalidateCachedFile(String path) {
```
[the remark this answers is not in the document]
Comment (huanghua78) [reply dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:356]: Yes. I forgot to delete it.

**Comment by huanghua78** [comment]
Thanks to @dbw9580 for reviewing this PR and adding implementation.

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by huanghua78** [comment]
alluxio-bot, merge this please

**Comment by huanghua78** [comment]
alluxio-bot, merge this please

**Pull request merged**